server/interfaces/interfaces.go

```diff
@@ -449,6 +451,18 @@ type MetricsCollector interface {
 	ReadCount(ctx context.Context, counterName string) (int64, error)
 }
 
+// A KeyValStore allows for storing ephemeral values globally.
+//
+// No guarantees are made about durability of KEyValStores -- they may be
+// evicted from the backing store that maintains them (usually memcache or
+// redis), so they should *not* be used in critical path code.
+type KeyValStore interface {
+	SetByKey(ctx context.Context, key string, val []byte) error
+	GetByKey(ctx context.Context, key string) ([]byte, error)
+	SetProtoByKey(ctx context.Context, key string, msg proto.Message) error
+	GetProtoByKey(ctx context.Context, key string, msg proto.Message) error
```

Comment:
nit: these could be util methods so that implementations of `KeyValStore` don't need to implement proto serialization / deserialization -- similar to how `cachetools.go` has utility methods for reading/writing protos from CAS:

```go
// file: server/util/keyval.go
package keyval

import (
  "github.com/buildbuddy-io/buildbuddy/server/interfaces"
)

func GetProto(store interfaces.KeyValStore, key string, msg proto.Message) error {
  ...
}

func SetProto(store interfaces.KeyValStore, key string, msg proto.Message) error {
  ...
}
```

Reply:
This is neat, thanks.

Reply:
Done!